app/assets/javascripts/controllers/middleware_topology/middleware_topology_controller.js

```diff
@@ -112,8 +150,25 @@ function MiddlewareTopologyCtrl($scope, $http, $interval, $location, topologySer
             default :
                 return { x: -9, y: -9, height: 18, width: 18, r: 17};
         }
+    }
 
-    };
+    self.getTextPosition = function getTextPosition(d) {
+        switch (d.item.kind) {
+            case "MiddlewareDatasource" :
+                return { x: 0, y: 8};
+            default :
+                return { x: 26, y: 24};
+        }
+    }
+
+    self.getGlyphFeatures = function getGlyphFeatures(d) {
+        switch (d.item.kind) {
+            case "MiddlewareDatasource" :
```

Comment:
Yeah, this makes sense. I was considering it, but then went with the faster solution.